src/safeposix/syscalls/net_calls.rs

```diff
@@ -902,4 +902,168 @@ impl Cage {
         }
         return retval; //package out fd_set?
     }
+
+    pub fn getsockopt_syscall(self, fd: i32, level: i32, optname: i32) -> i32 {
+
+        let mut fdtable = self.filedescriptortable.write().unwrap();
+        
+        if let Some(wrappedfd) = fdtable.get(&fd) {
+            let mut filedesc = wrappedfd.write().unwrap();
+            if let Socket(socketobj) = &mut *filedesc {
+                //checking that we recieved SOL_SOCKET\
+                match level {
+                    SOL_UDP => {
+                        return syscall_error(Errno::EOPNOTSUPP, "getsockopt", "UDP is not supported for getsockopt");
+                    }
+                    SOL_TCP => {
+                        return syscall_error(Errno::EOPNOTSUPP, "getsockopt", "TCP options not remembered by getsockopt");
+                    }
+                    SOL_SOCKET => {
+                        let STOREDSOCKETOPTIONS: Vec<i32> = vec![SO_LINGER, // ignored
+                                                                SO_KEEPALIVE, // ignored
+                                                                SO_SNDLOWAT, // ignored
+                                                                SO_RCVLOWAT, // ignored
+                                                                SO_REUSEPORT, // used to allow duplicate binds...
+                                                                SO_REUSEADDR];
+                        //if the option is a stored binary option, just return it...
+                        if STOREDSOCKETOPTIONS.contains(&optname) {
+                            if socketobj.options & optname == optname {
+                                return 1;
+                            }
+                            return 0;
+                        }
```

Comment:
I think this would be better as a match statement with a single arm for `SO_LINGER | SO_KEEPALIVE |`...